v1/plugin/plugin_test.go

```diff
@@ -253,12 +253,12 @@ func newMockErrStreamer() *mockStreamer {
 	return &mockStreamer{err: errors.New("empty")}
 }
 
-func newMockStreamerStream(action func(chan []Metric, time.Duration)) *mockStreamer {
+func newMockStreamerStream(action func(chan []Metric, time.Duration, []Metric)) *mockStreamer {
 	return &mockStreamer{action: action}
 }
 
-func (mc *mockStreamer) doAction(t time.Duration) {
-	go mc.action(mc.outMetric, t)
+func (mc *mockStreamer) doAction(t time.Duration, mts []Metric) {
+	go mc.action(mc.outMetric, t, mts)
```

Comment:
**To reviewers**:
Adding this []Metric to `action` allows providing tests with a various number of metrics in a batch.  It is used:
 - to mock stream 1 metric from stream plugin [here](https://github.com/intelsdi-x/snap-plugin-lib-go/pull/77/files#diff-c2dae3bc9807e0d41c60574bc91e2010R125)
 - to mock stream 4 metrics from stream plugin [here](https://github.com/intelsdi-x/snap-plugin-lib-go/pull/77/files#diff-c2dae3bc9807e0d41c60574bc91e2010R163)

So, not only the duration of coming the next streaming portion but also what metrics are there - all this staff might be mocked.